libraries/rush-lib/src/logic/pnpm/PnpmOptionsConfiguration.ts

```diff
@@ -16,6 +16,16 @@ import schemaJson from '../../schemas/pnpm-config.schema.json';
  */
 export type PnpmStoreOptions = 'local' | 'global';
 
+/**
+ * Possible values for the `resolutionMode` setting in Rush's pnpm-config.json file.
+ * @remarks
+ * These modes correspond to PNPM's `resolution-mode` values, which are documented here:
+ *
+ * {@link https://pnpm.io/npmrc#resolution-mode }
+ * @public
+ */
+export type PnpmResolutionModeOptions = 'highest' | 'time-based' | 'lowest-direct';
```

Comment:
```suggestion
export type PnpmResolutionMode = 'highest' | 'time-based' | 'lowest-direct';
```

Reply:
Maybe we can stick with this naming `PnpmResolutionModeOptions`? The existing `PnpmStoreOptions` using the same naming convention

Reply:
`PnpmStoreOptions` is also a non-ideal name for that type. In almost all cases, when we call a type `___Options`, it's an object that contains named parameters for a function.

Actually, here's a PR to rename that type: https://github.com/microsoft/rushstack/pull/4320

Reply:
Thanks for the explanation. I will rename it.